libraries/chain/apply_context.cpp

```diff
@@ -172,9 +176,11 @@ void apply_context::exec()
 {
    _notified.emplace_back( receiver, action_ordinal );
    exec_one();
+   trx_context.action_id.increment();
    for( uint32_t i = 1; i < _notified.size(); ++i ) {
       std::tie( receiver, action_ordinal ) = _notified[i];
       exec_one();
+      trx_context.action_id.increment();
```

Comment:
Same comment.  Why is `action_id` being incremented? Is this directly related to Deep Mind? If not should it be pulled out into a separate PR?